csrc/scheduler/utils.cpp

```diff
@@ -269,17 +269,17 @@ void parallelizeAllLike(
     }
     for (const auto i : c10::irange(tv->domain()->domain().size())) {
       auto ca_id = ca_map.getConcreteMappedID(
-          tv->axis(i), IdMappingMode::PERMISSIVE_RESIZE);
+          tv->axis((int)i), IdMappingMode::PERMISSIVE_RESIZE);
```

Comment:
Hmm, I feel this is a little too tedious as this is a pretty common pattern. Any other workaround than just casting to `int`? We could cast `tv->domain()->domain().size()` to int so that `i` becomes `int`, which may be slightly better? 

CC: @zasdfgbnm

Reply:
I believe `axis(int)` is still fine as the issue is signed vs unsigned, although I don't see any specific reason to leave it as `int` when everything else is changed to `int64_t`. 

How about making `TensorView::nDims()` to return `int64_t`? Here, we can replace `tv->domain()->domain().size()` with `tv->nDims()`, and if that returns `int64_t`, I believe we shouldn't need any explicit cast.

Reply:
Are you saying that this is a valid code?
```C++
void f(int);
int64_t i = 1l;
f(i);
```
I thought this was invalid. I thought we must do `f((int)i)` or change the signature of `f` to use `int64_t`, but I might be wrong.

On the other hand, I agree with changing `TensorView::nDims` to return `int64_t`.

Reply:
Hmm, sorry, that wasn't the case:

```
$ cat foo.cpp                                                                                                                                         
#include <cstdint>

void foo(int i);

void bar() {
  int64_t i = 1l;
  foo(i);
}
$ clang-tidy-13 foo.cpp                                                                                                                               
6 warnings generated.
/raid/nmaruyama/debug3/nvfuser/foo.cpp:7:7: error: narrowing conversion from 'int64_t' (aka 'long') to signed type 'int' is implementation-defined [bugprone-narrowing-conversions,cppcoreguidelines-narrowing-conversions,-warnings-as-errors]
  foo(i);
      ^
Suppressed 4 warnings (4 in non-user code).
Use -header-filter=.* to display errors from all non-system headers. Use -system-headers to display errors from system headers as well.
1 warning treated as error
```

Looks like we also need to change `TensorView::axis(int)` to `TensorView::axis(int64_t)`

Reply:
By doing so, we can ensure they use the same type, is that the main benefit?

Reply:
@naoyam Yes, that's the main benefit.

Reply:
I don't have a strong preference, but just using `int64_t` everywhere sounds just fine to me. Using a named type like `DimIdx_t` would allow us to easily change the actual type, but I don't expect that's going to happen.